src/datepicker/datepicker.js

```diff
@@ -629,7 +629,17 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
         scope.$parent.$watch(getAttribute, function(value) {
           scope.watchData[key] = value;
           if (key === 'minDate' || key === 'maxDate') {
-            cache[key] = new Date(value);
+            var date = parseDate(value);
+            if (!date) {
+              date = new Date(value);
+              if (angular.isDate(date) && !isNaN(date)) {
+                var dateArray = date.toJSON().split('T')[0].split('-');
+                date = new Date(+dateArray[0], +dateArray[1] - 1, +dateArray[2]);
```

Comment:
Oh, man. Sorry. I was on the wrong page with you. I was going for the edge-case support and incorrectly inferred that you still wanted to handle the edge-case when you added the `needs work` label. I don't know how the other way would be coded; my (not to strong) preference is to support this edge case.

If we decided to still allow literals, but required them to match the dateFormat, then we could just do a string check on the value here and return something like undefined or an invalid date in that case to just have it not work -- maybe that's what you are thinking?